sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -82,6 +82,17 @@ object AggregateUtils {
     }
   }
 
+  // This can be removed when all of our sorting is stable by default
+  // https://github.com/NVIDIA/spark-rapids/issues/9275
+  def isAllowedForUnstableSortedAgg(aggExprs: Seq[AggregateExpression]): Boolean = {
+    aggExprs.map(e => e.aggregateFunction).forall {
+      // These aggregations depend on the input order, so we cannot do an out of core sort and
+      // have it match.
+      case _: First | _: Last | _: CollectList => false
```

Comment:
Do we need this?  Worried about causing OOMs when they didn't exist before, and not sure this is technically necessary if Spark doesn't guarantee ordering due to the radix sort thing.